### PR TITLE
Use Eclipse Temurin instead of AdoptOpenJDK

### PR DESCRIPTION
**Pull request by ta7uw**
Motivation:

Update the URLs of OpenJDK in the documents because the new OpenJDK binaries from AdoptOpenJDK are no longer distributed and have been migrated to Eclipse Temurin.
It seems that this project have already used Eclipse Temurin on CI.

Modifications:

- Update the documents and URLs

Result:

- The URLs refer to Eclipse Temurin

### CODE REVIEW

File: site/src/pages/community/developer-guide.mdx [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ## Build requirements
 
-- [OpenJDK 11-17](https://adoptopenjdk.net/archive.html?variant=openjdk17&jvmVariant=hotspot)
+- [OpenJDK 11-17](https://adoptium.net/)
```
Comment (jrhee17) [inline site/src/pages/community/developer-guide.mdx:5]: Since 19 can also be used 😅 

```suggestion
- [OpenJDK 11-19](https://adoptium.net/)
```

**Comment by KarboniteKream** [comment]
Not completely related to this PR, but I think we can update this Java version in the documentation as well: https://github.com/line/armeria/blob/master/site/src/pages/docs/setup.mdx?plain=1#L11

**Review by jrhee17 — APPROVED** [review]
Thanks for the cleanup @ta7uw 🙇 🙇 🙇

**Review by ikhoon — APPROVED** [review]
Thanks!

**Review by minwoox — APPROVED** [review]
Thanks, @ta7uw and @KarboniteKream!

**Pull request merged**